fhir-search/src/main/java/com/ibm/fhir/search/util/SearchUtil.java

```diff
@@ -727,38 +728,51 @@ public static FHIRSearchContext parseQueryParameters(Class<?> resourceType,
+        for (Entry<String, List<String>> entry : queryParameters.entrySet()) {
+            String name = entry.getKey();
+            if (isSearchSingletonParameter(name) && entry.getValue().size() > 1) {
+                manageException("Search parameter '" + name + "' is specified multiple times", lenient);
+            }
         }
-        HashSet<String> resourceTypes = new HashSet<>();
-        if (Resource.class.equals(resourceType)) {
+
+        // Check for unsupported uses of _include/_revinclude
+        if (queryParameters.containsKey(SearchConstants.INCLUDE)
+                || queryParameters.containsKey(SearchConstants.REVINCLUDE)) {
+            // Make sure _sort is not present with _include and/or _revinclude.
+            // TODO: do we really need to forbid this?
+           if (queryParameters.containsKey(SearchConstants.SORT)) {
+                throw SearchExceptionUtil.buildNewInvalidSearchException(
+                        "_sort search result parameter not supported with _include or _revinclude.");
+            }
             // Because _include and _revinclude searches all require certain resource type modifier in
             // search parameter, so we just don't support it.
-            if (queryParameters.containsKey(SearchConstants.INCLUDE)
-                        || queryParameters.containsKey(SearchConstants.REVINCLUDE)) {
+            if (Resource.class.equals(resourceType)) {
                 throw SearchExceptionUtil.buildNewInvalidSearchException(
                         "system search not supported with _include or _revinclude.");
             }
+        }
 
-            if (queryParameters.containsKey(SearchConstants.RESOURCE_TYPE)) {
-                for (String resTypes : queryParameters.get(SearchConstants.RESOURCE_TYPE)) {
-                    List<String> tmpResourceTypes = Arrays.asList(resTypes.split("\\s*,\\s*"));
-                    for (String resType : tmpResourceTypes) {
-                        if (ModelSupport.isResourceType(resType)) {
-                            resourceTypes.add(resType);
-                        } else {
-                            manageException("_type search parameter has invalid resource type:" + resType, lenient);
-                            continue;
-                        }
+        // Check for unsupported uses of _type
+        if (queryParameters.containsKey(SearchConstants.RESOURCE_TYPE)) {
+            if (Resource.class.equals(resourceType)) {
+                // Only first value is used, which matches behavior of other parameters that are supposed to be specified at most once
+                String resTypes = queryParameters.get(SearchConstants.RESOURCE_TYPE).get(0);
+                List<String> tmpResourceTypes = Arrays.asList(resTypes.split("\\s*,\\s*"));
+                for (String resType : tmpResourceTypes) {
+                    if (ModelSupport.isResourceType(resType)) {
```

Comment:
I realize you didn't really change this line, but I wonder if `isResourceType("Resource")` returns true.  if so, is `_type=Resource` valid?   maybe we should lock it down to just concrete types (no abstract types).  maybe a separate issue...

Reply:
Updated to only accept concrete types.